pkg/model/trace/combine.go

```diff
@@ -90,9 +95,17 @@ func (c *Combiner) ConsumeWithFinal(tr *tempopb.Trace, final bool) (int, error)
 				}
 			}
 		}
-		return spanCount, c.sizeError()
+		moveForward, maxSizeErr := c.canContinue()
+		if moveForward {
+			return spanCount, nil
+		}
+		return spanCount, maxSizeErr
 	}
 
+	// Do not combine more spans for now
```

Comment:
i think this short circuit logic should go right after the if tr == nil check. it makes it clearer if these kinds of checks are all at the top of the func